app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,42 @@
+require 'purchase/presenters/new_purchase_presenter'
+require 'purchase/presenters/show_purchase_presenter'
+
+class PurchasesController < ApplicationController
+
+  def new
+    item = Item.find(params[:id])
+    purchase = Purchase.new
+    @presenter = NewPurchasePresenter.new(purchase, item)
+  end
+
+  def create
+    purchase = Purchase.new(email: buyer_email, item_id: params[:id])
+    Item.transaction do
+      item = Item.lock.find(params[:id])
+      @presenter = NewPurchasePresenter.new(purchase, item)
+      if purchase.save
+        item.update_attributes(available: false)
```

Comment:
What if someone else has already purchased the item between when my page was rendered and when I submit the request to purchase? You've introduced a lock, but aren't actually using it to validate that the item is still available.

You should also provide a nicer user experience if you find that the item has already been purchased.